cli/boards.go

```diff
@@ -0,0 +1,234 @@
+	return nil
+}
+
+func (c *appClient) uploadBoardDefsFile(
+	name string,
+	version string,
+	orgID string,
+	jsonFile *os.File,
+) (*packagepb.CreatePackageResponse, error) {
+	if err := c.ensureLoggedIn(); err != nil {
+		return nil, err
+	}
+	ctx := c.c.Context
+
+	// Create an archive tar.gz file (required for packages).
+	file, err := CreateArchive(jsonFile)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating archive")
+	}
+
+	// The board defs packages are small and never expected to be larger than the upload chunk size,
+	// so we are sending in one chunk.
+	// If the file is too big, return error.
+	if file.Len() > uploadChunkSize {
+		return nil, fmt.Errorf("file is too large, must be under %d bytes", uploadChunkSize)
+	}
+
+	stream, err := c.packageClient.CreatePackage(ctx)
+	if err != nil {
+		return nil, errors.Wrapf(err, "error starting CreatePackage stream")
+	}
+
+	stats, err := jsonFile.Stat()
+	if err != nil {
+		return nil, err
+	}
+	boardDefsFile := []*packagepb.FileInfo{{Name: name, Size: uint64(stats.Size())}}
+
+	packageInfo := &packagepb.PackageInfo{
+		OrganizationId: orgID,
+		Name:           name,
+		Version:        version,
+		Type:           packagepb.PackageType_PACKAGE_TYPE_BOARD_DEFS,
+		Files:          boardDefsFile,
+		Metadata:       nil,
+	}
+
+	// send the package requests
+	var errs error
+	if err := sendPackageRequests(stream, file, packageInfo); err != nil {
+		errs = multierr.Combine(errs, errors.Wrapf(err, "error syncing package"))
+	}
+
+	// close the stream and receive a response when finished.
+	resp, err := stream.CloseAndRecv()
+	if err != nil {
+		errs = multierr.Combine(errs, errors.Wrapf(err, "received error response while syncing package"))
+	}
+	if errs != nil {
+		return nil, errs
+	}
+
+	return resp, nil
+}
+
+// helper function to check if a package with this version already exists.
+func (c *appClient) lookForPackage(ctx context.Context, orgID, name, version string) error {
+	// the packageID is the orgid/name
+	packageID := fmt.Sprintf("%s/%s", orgID, name)
+
+	req := packagepb.GetPackageRequest{
+		Id:      packageID,
+		Version: version,
+	}
+
+	_, err := c.packageClient.GetPackage(ctx, &req)
+
+	if !strings.Contains(err.Error(), "package not found") {
+		return fmt.Errorf("a package with name %s and version %s already exists", name, version)
+	}
+	return nil
+}
+
+func sendPackageRequests(stream packagepb.PackageService_CreatePackageClient,
+	f *bytes.Buffer, packageInfo *packagepb.PackageInfo,
+) error {
+	req := &packagepb.CreatePackageRequest{
+		Package: &packagepb.CreatePackageRequest_Info{Info: packageInfo},
+	}
+	// first send the packageinfo.
+	if err := stream.Send(req); err != nil {
+		return err
+	}
+
+	//nolint:errcheck
+	defer stream.CloseSend()
+
+	// Now send the file contents.
+	req = &packagepb.CreatePackageRequest{
+		Package: &packagepb.CreatePackageRequest_Contents{Contents: f.Bytes()},
+	}
+
+	if err := stream.Send(req); err != nil {
+		return err
+	}
```

Comment:
[opt/low prio/ feel free to resolve/ probably doesn't matter] If you want to support larger files, I don't think its too hard: (I took this from another project I worked on)
```go
for i := 0; i < len(bytes); {
		if ctx.Err() != nil {
			return ctx.Err()
		}
		numBytesToSend := uploadChunkSize
		if i+numBytesToSend > len(bytes) {
			numBytesToSend = len(bytes) - i
		}
		next := bytes[i : i+numBytesToSend]
		i += numBytesToSend

		nextReq := req = &packagepb.CreatePackageRequest{
		    Package: &packagepb.CreatePackageRequest_Contents{Contents: f.Bytes()},
	        }

		if err := stream.Send(nextReq); err != nil {
			return err
		}
	}
```
(might have off by one errs -- haven't checked)